### DISCUSSION TITLE
Problema con Caddy

### DISCUSSION THREAD

**Discussion by mbriacuni** [indomus/forum#1883] [GENERALE - Discussioni generiche]
Salve, qualche mese fa ho seguito con successo la vostra guida per installare Caddy su raspberry os con docker e fino a qualche giorno fa funzionava senza problemi. 
Probabilmente da un paio di giorni (ho avuto casini a lavoro e non ho potuto controllare effettivamente prima di oggi) già non funziona e ho provato a effettuare un aggiornamento del container, senza risultati. 
Ho pensato bene di rimuovere il tutto e ricominciare da capo, probabilente sbagliando, e ora non riesco più neanche a completare la creazione del container. 

```
pi@raspberrypi:~ $ docker-compose up caddy
Creating caddy ... done
Attaching to caddy
caddy             | {"level":"info","ts":1662200837.432164,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
caddy             | {"level":"warn","ts":1662200837.4385285,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":3}
caddy             | {"level":"info","ts":1662200837.4490674,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]}
caddy             | {"level":"info","ts":1662200837.4502308,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
caddy             | {"level":"info","ts":1662200837.450355,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
caddy             | {"level":"info","ts":1662200837.454746,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/data/caddy"}
caddy             | {"level":"info","ts":1662200837.4553645,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["*****MIO DOMINO*****"]}
caddy             | {"level":"info","ts":1662200837.4571934,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x4000372cb0"}
caddy             | {"level":"info","ts":1662200837.473307,"logger":"tls","msg":"finished cleaning storage units"}
caddy             | {"level":"info","ts":1662200837.4783835,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
caddy             | {"level":"info","ts":1662200837.4784596,"msg":"serving initial configuration"}

```

quando avvio il comando da terminale si ferma sempre a questo punto e non so come procedere.

questa è la porzione del file docker-compose:

```
  caddy:
    image: caddy:latest
    container_name: caddy
    ports:
    - "8123:8123"
    - "80:80"
    - "443:443"
    environment:
    - "TZ=Europe/Rome"
    volumes:
    - "/home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile"
    - "/home/pi/caddy/data:/data"
    restart: always
```

Vi ringrazio in anticipo per l'aiuto

**Comment by snakuzzo** [comment]
Guarda che il log non segnala alcun errore. È tutto ok...i certificati li hai già, quindi è a posto così.

Il comando da dare è...

```
docker-compose up -d caddy
```

Se lo lanci senza il `-d` lo avvii in foreground e nel momento che dai CTRL+C il container si ferma.

**Reply by mbriacuni** [reply]
Ok, penso sia abbastanza chiaro che di linux ne capisco poco. 
Ho completato nuovamente tutta la configurazione, ma non riesco ancora ad accedere ad HA in remoto. Ho provato a collegarmi a `https://miodominio` senza specificare la porta e mi apre la pagina di configurazione del router (se accedo da rete locale) per cui penso che sia tutto apposto per quanto riguarda caddy. 

Nel configuration.yaml ho già questa porzione di codice

```
http:
  use_x_forwarded_for: true
  trusted_proxies: 
    - 172.0.0.0/8
```

non so se il problema possa dipendere dall'indirizzo del proxy errato, ma nel log di HA non ho trovato nulla riguardante quell'errore menzionato nella guida

**Reply by mbriacuni** [reply]
Ok, è la volta buona che faccio volare la Vodafone station dall finesta....... Ho eliminato e reinserito la regola di port forwading e ora si collega perfettamente...... Grazie per l'attenzione e scusa il disturbo.

**Reply by snakuzzo** [reply]
Ok. Bene 
Ma Linux in tutto ciò non c'entra niente.
Sarebbe stato lo stesso su altri OS

**Reply by mbriacuni** [reply]
Si, mi riferivo al fatto che inizialmente mi sono impantanato in una fesseria senza neanche capire che il mio problema era che non avviavo il comando come daemon
